cpp/oneapi/dal/backend/primitives/common.hpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <cstdint>
+
+namespace oneapi::dal::backend::primitives {
+
+enum class layout : int { row_major = 0, col_major = 1 };
```

Comment:
We do already have `data_layout` in https://github.com/oneapi-src/oneDAL/blob/master/cpp/oneapi/dal/table/common.hpp
Need to think, do we want to have a dependency on the `table` component here or not